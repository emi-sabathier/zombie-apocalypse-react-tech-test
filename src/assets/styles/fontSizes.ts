type FontSizesType = {
    [key: string]: string;
};

export const fontSizes: FontSizesType = {
    small: '1.2rem',
    regular: '1.4rem',
    large: '2rem',
    xlarge: '2.6rem',
    xxlarge: '5rem',
};
