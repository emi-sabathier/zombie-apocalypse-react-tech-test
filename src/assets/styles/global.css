@import url('https://fonts.googleapis.com/css2?family=Poppins:wght@500;600;700&display=swap');
@import url('https://fonts.googleapis.com/css2?family=Creepster&display=swap');

html {
  font-size: 62.5%;
}
body {
  background-color: #7209B7;
  font-size: 1.6rem;
  margin: 0;
  font-family: 'Poppins', sans-serif;
}

h1 {
  font-family: 'Creepster', sans-serif;
}

p {
  margin-block-start: 0;
  margin-block-end: 0;
}

h1, h2 {
  margin-block-start: 0;
  margin-block-end: 0;
  font-weight: normal;
}

button {
  background: none;
  border: none;
  padding: 0;
}
