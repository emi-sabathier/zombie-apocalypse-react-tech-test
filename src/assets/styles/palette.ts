type PaletteType = {
    [key: string]: string;
};

export const palette: PaletteType = {
    headerBackground: '#3A0CA3',
    white: '#FFF',
};
