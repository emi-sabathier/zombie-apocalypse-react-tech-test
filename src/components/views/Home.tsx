// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-nocheck
import React, { ReactElement } from 'react';
import { UIContainer } from '../shared/UIContainer/UIContainer';
import { useDocumentTitle } from '../../hooks/useDocumentTitle';
import { UITeamMateCard } from '../shared/UITeamMateCard/UITeamMateCard';
import styled from 'styled-components';
import { useFetch } from '../../hooks/useFetch';
import { Spinner } from '../Spinner';

export function Home(): ReactElement {
    useDocumentTitle('Zombie Apocalypse - Accueil');
    const { dataList, isLoading } = useFetch('https://reqres.in/api/users?per_page=12');
    const { data } = dataList ?? [];
    console.log(data);
    return (
        <UIContainer>
            {isLoading ? (
                <Spinner />
            ) : (
                <CardsContainer>
                    {data.map(mate => (
                        <UITeamMateCard key={mate.id} mate={mate} />
                    ))}
                </CardsContainer>
            )}
        </UIContainer>
    );
}

const CardsContainer = styled.div`
    display: grid;
    grid-template-columns: repeat(4, 25%);
    justify-items: center;
`;
