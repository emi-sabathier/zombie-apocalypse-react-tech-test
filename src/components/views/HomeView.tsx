import React, { ReactElement } from 'react';
import { UIContainer } from '../shared/UIContainer/UIContainer';
import { useDocumentTitle } from '../../hooks/useDocumentTitle';
import { UIMateCard } from '../UIMateCard';
import styled from 'styled-components';
import { useFetch } from '../../hooks/useFetch';
import { Spinner } from '../Spinner';
import { Mate } from '../../Model/MateModel';
import { ResponseMatesListModel } from '../../Model/ResponseMatesListModel';
import { API_URL_LIST } from '../../constants/constants';

export function HomeView(): ReactElement {
    useDocumentTitle('Zombie Apocalypse - Accueil');
    const { error, response, isLoading } = useFetch<ResponseMatesListModel>(API_URL_LIST);
    return (
        <UIContainer>
            {isLoading && <Spinner />}
            {error && <p>Erreur</p>}
            <CardsContainer>
                {response?.data?.map((mate: Mate) => (
                    <UIMateCard key={mate.id} mate={mate} />
                ))}
            </CardsContainer>
        </UIContainer>
    );
}

const CardsContainer = styled.div`
    display: grid;
    grid-template-columns: repeat(4, 25%);
    justify-items: center;
`;
