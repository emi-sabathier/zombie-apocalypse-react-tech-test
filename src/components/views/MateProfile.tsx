import React from 'react';
import styled from 'styled-components';
import { useLocation } from 'react-router-dom';
import { useDocumentTitle } from '../../hooks/useDocumentTitle';
import { UIContainer } from '../shared/UIContainer/UIContainer';
import { palette } from '../../assets/styles/palette';
import { fontSizes } from '../../assets/styles/fontSizes';
import { UIAddDeleteMate } from '../shared/UIAddDeleteMate/UIAddDeleteMate';

const RADIUS = 5;
const WIDTH = '50%';
const PADDING = 10;

export function MateProfile() {
    const { state } = useLocation();
    useDocumentTitle(`Profil de ${state.first_name} ${state.last_name}`);

    return (
        <UIContainer>
            <ProfileContainer>
                <Avatar src={`${state.avatar}`} alt={`${state.first_name} ${state.last_name} avatar`} />
                <DetailsContainer>
                    <NameTitle>
                        {state.first_name} {state.last_name}
                    </NameTitle>
                    <a href={`mailto:${state.email}`}>{state.email}</a>
                    <ButtonsContainer>
                        <UIAddDeleteMate />
                    </ButtonsContainer>
                </DetailsContainer>
            </ProfileContainer>
        </UIContainer>
    );
}

const ButtonsContainer = styled.div`
    flex: 1;
    align-items: flex-end;
    display: flex;
`;
const DetailsContainer = styled.div`
    padding: 0 ${PADDING}px;
    display: flex;
    flex-direction: column;
    flex: 1;
    align-items: center;
`;

const NameTitle = styled.h2`
    font-size: ${fontSizes.xlarge};
    font-weight: 700;
`;

const ProfileContainer = styled.section`
    width: ${WIDTH};
    background-color: ${palette.white};
    border-radius: ${RADIUS}px;
    padding: ${PADDING}px;
    display: flex;
`;

const Avatar = styled.img`
    border-radius: ${RADIUS}px;
    width: 50%;
`;
