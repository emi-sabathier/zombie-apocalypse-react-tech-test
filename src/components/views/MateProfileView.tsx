import React from 'react';
import styled from 'styled-components';
import { UIContainer } from '../shared/UIContainer/UIContainer';
import { palette } from '../../assets/styles/palette';
import { fontSizes } from '../../assets/styles/fontSizes';
import { useFetch } from '../../hooks/useFetch';
import { useParams } from 'react-router-dom';
import { Spinner } from '../Spinner';
import { ResponseSingleMateModel } from '../../Model/ResponseMatesListModel';
import { useIsMateExistsInStore } from '../../hooks/useIsMateExistsInStore';
import { API_URL_SINGLE_MATE } from '../../constants/constants';
import { UIAddDeleteMate } from '../shared/UIAddDeleteMate/UIAddDeleteMate';
import { useDocumentTitle } from '../../hooks/useDocumentTitle';
import { IsMateSelectedProps } from '../UIMateCard';
import { dictionary } from '../../assets/dictionary';

const RADIUS = 5;
const WIDTH = '50%';
const PADDING = 10;

export function MateProfileView() {
    const { id } = useParams();
    const { error, response, isLoading } = useFetch<ResponseSingleMateModel>(`${API_URL_SINGLE_MATE}${id}`);
    const isMateExists = useIsMateExistsInStore(Number(id));
    const res = response?.data ?? {
        id: 0,
        avatar: '',
        email: '',
        first_name: '',
        last_name: '',
    };

    useDocumentTitle(`Profil de ${response?.data?.first_name} ${response?.data?.last_name}`);

    return (
        <UIContainer>
            {error && (
                <BackContainer>
                    <p>{dictionary.error.text}</p>
                </BackContainer>
            )}
            {isLoading && !error && <Spinner />}
            {!isLoading && (
                <ProfileContainer isMateSelected={isMateExists}>
                    <Avatar
                        src={`${response?.data?.avatar}`}
                        alt={`${response?.data?.first_name} ${response?.data?.last_name} avatar`}
                    />
                    <DetailsContainer>
                        <NameTitle>
                            {response?.data?.first_name} {response?.data?.last_name}
                        </NameTitle>
                        <a href={`mailto:${response?.data?.email}`}>{response?.data?.email}</a>
                        <ButtonsContainer>
                            <UIAddDeleteMate mate={res} />
                        </ButtonsContainer>
                    </DetailsContainer>
                </ProfileContainer>
            )}
        </UIContainer>
    );
}

const BackContainer = styled.div`
    display: flex;
    justify-content: center;
    align-items: center;
    height: 70vh;
    color: ${palette.white};
    font-size: ${fontSizes.xlarge};
    font-weight: 700;
    flex-direction: column;
`;

const ProfileContainer = styled.section<IsMateSelectedProps>`
    border: ${props => (props.isMateSelected ? '5px solid red' : '5px solid white')};
    width: ${WIDTH};
    background-color: ${palette.white};
    border-radius: ${RADIUS}px;
    padding: ${PADDING}px;
    display: flex;
    @media (max-width: 425px) {
        align-items: center;
        flex: 1;
        flex-direction: column;
    }
`;

const ButtonsContainer = styled.div`
    flex: 1;
    align-items: flex-end;
    display: flex;
`;
const DetailsContainer = styled.div`
    padding: 0 ${PADDING}px;
    display: flex;
    flex-direction: column;
    flex: 1;
    align-items: center;
`;

const NameTitle = styled.h2`
    font-size: ${fontSizes.xlarge};
    font-weight: 700;
`;
const Avatar = styled.img`
    border-radius: ${RADIUS}px;
    width: 50%;
`;
