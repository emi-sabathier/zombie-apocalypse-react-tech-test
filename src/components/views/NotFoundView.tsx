import React from 'react';
import styled from 'styled-components';
import { fontSizes } from '../../assets/styles/fontSizes';
import { palette } from '../../assets/styles/palette';

export function NotFoundView() {
    return (
        <Container>
            <P>La route n&apos;existe pas</P>
        </Container>
    );
}

export const Container = styled.div`
    display: flex;
    justify-content: center;
    align-items: center;
    height: 70vh;
    color: ${palette.white};
    font-size: ${fontSizes.xlarge};
    font-weight: 700;
`;

export const P = styled.p`
    font-size: ${fontSizes.xlarge};
`;
