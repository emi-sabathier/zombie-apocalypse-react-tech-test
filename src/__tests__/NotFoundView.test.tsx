import React from 'react';
import { Container, NotFoundView, P } from '../components/views/NotFoundView';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import { palette } from '../assets/styles/palette';
import { fontSizes } from '../assets/styles/fontSizes';

describe('NotFoundView', () => {
    it('should display error message', () => {
        render(<NotFoundView />);
        const linkElement = screen.getByText("La route n'existe pas");
        expect(linkElement).toBeInTheDocument();
    });
    it('should match P CSS', () => {
        const tree = renderer.create(<P />).toJSON();
        expect(tree).toHaveStyleRule('font-size', '2.6rem');
    });
    it('should match Container CSS', () => {
        const tree = renderer.create(<Container />).toJSON();
        expect(tree).toHaveStyleRule('display', 'flex');
        expect(tree).toHaveStyleRule('justify-content', 'center');
        expect(tree).toHaveStyleRule('align-items', 'center');
        expect(tree).toHaveStyleRule('height', '70vh');
        expect(tree).toHaveStyleRule('color', `${palette.white}`);
        expect(tree).toHaveStyleRule('font-size', `${fontSizes.xlarge}`);
        expect(tree).toHaveStyleRule('font-weight', '700');
    });
});
