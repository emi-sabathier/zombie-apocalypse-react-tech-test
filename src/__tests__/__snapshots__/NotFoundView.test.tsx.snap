// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundView jpp 1`] = `
<p
  className="sc-gswNZR gBlwrG"
/>
`;
